### PR TITLE
[WIP] Add tests

### PR DESCRIPTION
**Pull request by blink1073**
```
SUMMARY:
✔ 189 tests completed
--------------|----------|----------|----------|----------|----------------|
File          |  % Stmts | % Branch |  % Funcs |  % Lines |Uncovered Lines |
--------------|----------|----------|----------|----------|----------------|
 lib/         |    97.04 |    84.77 |     99.3 |    96.99 |                |
  index.js    |      100 |      100 |      100 |      100 |                |
  menu.js     |    97.21 |    85.61 |      100 |    97.16 |... 808,827,830 |
  menubar.js  |    95.45 |    76.77 |    97.37 |    95.41 |... 536,539,573 |
  menubase.js |    98.55 |    88.89 |      100 |    98.48 |            114 |
  menuitem.js |      100 |      100 |      100 |      100 |                |
--------------|----------|----------|----------|----------|----------------|
All files     |    97.04 |    84.77 |     99.3 |    96.99 |                |
--------------|----------|----------|----------|----------|----------------|
```

### CODE REVIEW

File: test/src/menubar.ts [hunk 1]
```diff
@@ -0,0 +1,731 @@
...
+
+    it('should equal `p-MenuBar`', () => {
+      expect(MENU_BAR_CLASS).to.be('p-MenuBar');
+    });
+
+  });
+
+  describe('CONTENT_CLASS', () => {
+
+    it('should equal `p-Menu-content`', () => {
```
Comment (blink1073) [inline test/src/menubar.ts]: Several of these are defined in both `menu.ts` and `menubar.ts` as different strings: `CONTENT`, `MENU_ITEM`, `ICON`, `TEXT`

File: test/src/menu.ts [hunk 2]
```diff
@@ -0,0 +1,966 @@
...
+function triggerMouseEvent(node: HTMLElement, eventType: string, options: any={}) {
+  options.bubbles = true;
+  var clickEvent = new MouseEvent(eventType, options);
+  node.dispatchEvent(clickEvent);
+}
+
+
+function triggerKeyEvent(node: HTMLElement, eventType: string, options: any={}) {
+  // cannot use KeyboardEvent in Chrome because it seys keyCode = 0
+  var event = document.createEvent('Event');
```
Comment (blink1073) [inline test/src/menu.ts]: `keyCode` is deprecated, which is probably why Chrome forces it to zero.  Should we be using `key` instead?

File: test/src/menu.ts [hunk 3]
```diff
@@ -0,0 +1,961 @@
...
+function triggerMouseEvent(node: HTMLElement, eventType: string, options: any={}) {
+  options.bubbles = true;
+  var clickEvent = new MouseEvent(eventType, options);
+  node.dispatchEvent(clickEvent);
+}
+
+
+function triggerKeyEvent(node: HTMLElement, eventType: string, options: any={}) {
+  // cannot use KeyboardEvent in Chrome because it sets keyCode = 0
+  var event = document.createEvent('Event');
```
Comment (blink1073) [inline test/src/menu.ts]: `keyCode` is deprecated, which is probably why Chrome forces it to zero.  Should we be using `key` instead?

Comment (sccolbert) [inline test/src/menu.ts]: While deprecated, `keyCode` is the only thing that actually works on all browsers. Chrome does not support `key`.

Comment (blink1073) [inline test/src/menu.ts]: Good times.

**Comment by blink1073** [comment]
:tada:, that was a beast to test.

**Pull request merged**